Hazel.UnitTests/Dtls/DtlsConnectionTests.cs

```diff
@@ -313,6 +313,81 @@ public void TestMalformedConnectionData()
             }
         }
 
+
+        [TestMethod]
+        public void TestReorderedCertFragmentsConnects()
+        {
+            IPEndPoint captureEndPoint = new IPEndPoint(IPAddress.Loopback, 27511);
+            IPEndPoint listenerEndPoint = new IPEndPoint(IPAddress.Loopback, 27510);
+
+            bool serverConnected = false;
+            bool serverDisconnected = false;
+            bool clientDisconnected = false;
+
+            Semaphore signal = new Semaphore(0, int.MaxValue);
+
+            var logger = new TestLogger("Throttle");
+
+            using (SocketCapture capture = new SocketCapture(captureEndPoint, listenerEndPoint, logger))
+            using (DtlsConnectionListener listener = new DtlsConnectionListener(2, new IPEndPoint(IPAddress.Any, listenerEndPoint.Port), new TestLogger("Server")))
+            using (DtlsUnityConnection connection = new DtlsUnityConnection(new TestLogger("Client "), captureEndPoint))
+            {
+                Semaphore listenerToConnectionThrottle = new Semaphore(0, int.MaxValue);
+                capture.SendToLocalSemaphore = listenerToConnectionThrottle;
+                Thread throttleThread = new Thread(() => {
+                    // HelloVerifyRequest
+                    while (capture.PacketsForLocalCount == 0) Thread.Sleep(10);
+                    Assert.AreEqual(1, capture.PacketsForLocalCount);
+                    listenerToConnectionThrottle.Release(1);
+
+                    // ServerHello, Server Certificate (Fragment)
+                    // Server Cert
+                    // ServerKeyExchange, ServerHelloDone
+                    while (capture.PacketsForLocalCount < 3) Thread.Sleep(10);
+                    Assert.AreEqual(3, capture.PacketsForLocalCount);
+                    capture.ReversePacketsForLocal();
```

Comment:
So yeah, continuing the other comment, it's this line I've been thinking about - while it does indeed flip the order of the two cert packets, I think the reversal means the 3rd packet, `ServerKeyExchange, ServerHelloDone`-packet, is the one that arrives first? If so, technically it might be testing the un-ordered arrival of that packet rather than that of the cert packets specifically

I definitely understand the choice of creating the Reverse method as to not swell the scope of this PR too much, though the wishlist item would be to be able to specify the order exactly

Might be too much to do that for now, but wondering if there's an easy move we can make to just flip the two cert packets?